12-target-details.tex

```diff
@@ -5,6 +5,144 @@ \section{Javascript}
+To clarify, in most cases we have to add a \ic{<script>}-tag that links to the external library manually to the HTML-document.
+
+
+An example of an extern class is the \href{https://github.com/HaxeFoundation/haxe/blob/development/std/js/JQuery.hx#L83}{jQuery class} of the Standard Library. 
+To illustrate here is a part of this extern class:
+
+\begin{lstlisting}
+@:initPackage
+extern class JQuery implements ArrayAccess<Element> {
+	function addClass( className : String ) : JQuery;
+	function removeClass( ?className : String ) : JQuery;
+	function hasClass( className : String ) : Bool;
+	
+	@:overload(function(html:String):JQuery{})
+	@:overload(function(html:JQuery):JQuery{})
+	function html() : String;
+..
+\end{lstlisting}
+
+Note that (only) in extern classes functions can be overloaded to accept different types of arguments and return values, using the \expr{@:overload} metatag.
+Using this extern, we can use jQuery like this:
+
+\begin{lstlisting}
+new JQuery("#my-div").addClass("brand-success").html("haxe is great!");
+\end{lstlisting}
+
+Beside externs, \tref{Typedefs}{type-system-typedef} can be a great way to name (or alias) complex Javascript or JSON structures.
+
+The package and class name of the extern should be in the same as defined in the external library. If that doesn't, rewrite the path of a class using \expr{@:native}.
+
+\begin{lstlisting}
+package my.application.media;
+
+@:native('external.library.media.video')
+extern class Video {
+..
+\end{lstlisting}
+
+From Haxe, it is possible to call an exposed functions thanks to the \expr{untyped} keyword. This can be useful in some cases if we don't want to write externs. Anything untyped that is valid syntax will generated as it is.
+
+\begin{lstlisting}
+untyped window.trackEvent("page1");  
+\end{lstlisting}
+
+Using the magic \expr{__js__} function we can inject pure Javascript code fragments into the output. This code can not be validated, so it accepts invalid and code in the output, which is error-prone.
+For example this writes a Javascript comment in the output.
+
+\begin{lstlisting}
+untyped __js__('// haxe is great!');
+\end{lstlisting}
+
+The standard compilation options also provides more Haxe sources to be added to the project:
+
+\begin{itemize}
+	\item To add a \tref{Haxelib library}{haxelib}, use \expr{-lib <library-name>}. There are many externs for popular native libraries.
+	\item To add another class path, use \expr{-cp <directory>}.
+	\item To force a whole package to be included in the project, use \expr{--macro include('mypackage')} that will include all the classes declared in the given package and subpackages. 
+\end{itemize}
+
+\subsection{Javascript target Metatags}
+\label{target-javascript-metatags}
+
+This is the list of Javascript specific metatags. See also the complete list of all \tref{Haxe built-in metatags}{cr-metadata}.
```

Comment:
"For more information, see also"...or "If you are looking for a complete...., see ,,,,"